pkg/controllers/work/applier.go

```diff
@@ -0,0 +1,105 @@
+		return nil, errorApplyAction, controller.NewAPIServerError(false, err)
+	}
+	klog.V(2).InfoS("Manifest apply succeeded", "gvr", gvr, "manifest", manifestRef)
+	return manifestRes, manifestServerSideAppliedAction, nil
+}
+
+// findConflictedWork checks if the manifest is owned by other placements which have configured different strategy.
+// It returns the first conflicted work it finds.
+func findConflictedWork(ctx context.Context, hubClient client.Client, namespace string, strategy *fleetv1beta1.ApplyStrategy, ownerRefs []metav1.OwnerReference) (*fleetv1beta1.Work, error) {
+	for _, ownerRef := range ownerRefs {
+		if ownerRef.APIVersion != fleetv1beta1.GroupVersion.String() || ownerRef.Kind != fleetv1beta1.AppliedWorkKind {
+			continue
+		}
+
+		// we need to check if the appliedWork is using the same strategy as the work
+		// Note, the resource could be already owned by the current work and apply strategy of the work may be changed
+		// too during the check.
+		// We'll return the user error to retry again.
+		work := &fleetv1beta1.Work{}
+		name := types.NamespacedName{Namespace: namespace, Name: ownerRef.Name}
+		if err := hubClient.Get(ctx, name, work); err != nil {
+			klog.ErrorS(err, "Failed to retrieve the work", "work", name)
+			if errors.IsNotFound(err) {
+				// The work may be just deleted by the time we are checking it and the ownerRef is already stale.
+				// Or it could be manually deleted by the user.
+				// Return the error to retry.
+				return nil, controller.NewExpectedBehaviorError(err)
+			}
+			return nil, controller.NewAPIServerError(true, err)
+		}
+		// TODO, could be removed once we have the defaulting webhook with fail policy.
+		// Make sure these conditions are met before moving
+		// * the defaulting webhook failure policy is configured as "fail".
+		// * user cannot update/delete the webhook.
+		defaulter.SetDefaultsWork(work)
+		if !equality.Semantic.DeepEqual(strategy, work.Spec.ApplyStrategy) {
+			return work, nil
+		}
+	}
+	return nil, nil
+}
+
+func validateOwnerReference(ctx context.Context, hubClient client.Client, namespace string, strategy *fleetv1beta1.ApplyStrategy, ownerRefs []metav1.OwnerReference) (ApplyAction, error) {
+	conflictedWork, err := findConflictedWork(ctx, hubClient, namespace, strategy, ownerRefs)
+	if err != nil {
+		return errorApplyAction, err
+	}
+	if conflictedWork != nil {
+		placement := conflictedWork.Labels[fleetv1beta1.CRPTrackingLabel]
+		err := fmt.Errorf("manifest is already managed by placement %s but with different apply strategy or the placement strategy is changed", placement)
+		klog.ErrorS(err, "Skip applying a manifest managed by another placement but with different apply strategy",
+			"conflictedWork", conflictedWork.Name, "conflictedPlacement", placement, "conflictedWorkApplyStrategy", conflictedWork.Spec.ApplyStrategy)
+		return applyConflictBetweenPlacements, controller.NewUserError(err)
+	}
+
+	// check if the existing manifest is managed by the work
```

Comment:
Hi Zhiying! Just a nit: would you mind updating the comments to be a bit more specific (e.g., -> `skip if the current resource is not derived from the work and co-ownership is disallowed`)?

Reply:
Also, do you think it would be necessary if we add a comment here noting that with the current `isManifestManagedByWork` behavior change, when the co-ownership is added afterwards, e.g., by a controller using on the user-side, all resource changes will cease to sync?